### PR TITLE
Set flags via deploy environment

### PR DESCRIPTION
**Pull request by Catacola**
## Description of the change

Update the flags.js file to return different sets of flags based on which stage of deployment the app is in. This will make it so we don't need to remember which features are enabled where when deploying.

To accomplish this, I transitioned us to store all of our frontend env vars in `.env-cmdrc`. The purpose of this change was to store all config variables in one file where they can be compared and examined easily.

Once this PR is approved and before landing, I will update the secrets zip in 1password.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [x] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: .env-cmdrc.example [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "REACT_APP_SENTRY_DSN": "{SENTRY_DSN}"
+  },
+  "e2e": {
+    "REACT_APP_DEPLOY_ENV": "dev",
+    "REACT_APP_API_URL": "{API_URL}",
+    "REACT_APP_AUTH_ENV": "{AUTH_ENV}",
+    "REACT_APP_FEEDBACK_URL": "{FEEDBACK_URL}",
+    "REACT_APP_INTERCOM_APP_ID": "{INTERCOM_APP_ID}",
+    "REACT_APP_SENTRY_ENV": "{SENTRY_ENC}",
+    "REACT_APP_SENTRY_DSN": "{SENTRY_DSN}"
```
Comment (daschi) [inline .env-cmdrc.example:18]: I think we'll also want to add: 
`"RUN_TESTS_HEADLESS": "true"`
and
`TEST_AUTH_USER`
`TEST_AUTH_PASSWORD`

File: .env-cmdrc.example [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "development": {
+    "REACT_APP_DEPLOY_ENV": "dev",
+    "REACT_APP_API_URL": "{API_URL}",
+    "REACT_APP_AUTH_ENV": "{AUTH_ENV}",
+    "REACT_APP_FEEDBACK_URL": "{FEEDBACK_URL}",
+    "REACT_APP_INTERCOM_APP_ID": "{INTERCOM_APP_ID}",
+    "REACT_APP_SENTRY_ENV": "{SENTRY_ENC}",
+    "REACT_APP_SENTRY_DSN": "{SENTRY_DSN}"
```
Comment (daschi) [inline .env-cmdrc.example:9]: Should we also add `REACT_APP_IS_DEMO="false"` to the development one?

**Review by daschi — APPROVED** [review]
A couple of suggestions but love this! This'll be so much easier to work with

**Review by jovergaag — APPROVED** [review]
🙏

**Pull request merged**